src/components/SearchBar/SearchBar.tsx

```diff
@@ -37,19 +33,31 @@ const SearchBar: React.ForwardRefRenderFunction<
     disabled = false,
     focused = false,
     label,
-    onClick,
+    error,
+    helperText,
+    onSearch,
     sx,
   } = props;
 
+  const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    if (onChange != undefined) onChange(event.target.value);
```

Comment:
I think `onChange` should not be optional. I know that you just added fixes for testing, but then here we should not need this check of `onChange != undefined` because it will always be defined. And I think same should happen to `onSearch`, since these two values will be mandatory. 

In general really nice work! Story testing I think would make our work easier. But anyways will need to get use to it and I think it will take some time for me 😄

Reply:
I agree they shouldn't be optional. I'm running into issues with the stories and interaction testing when I make them required though.
If they are required I have to assign e.g. onChange in the Template in .stories (if they are optional that happens automagically somehow). I can't seem to use hooks in stories so I can't give it a proper state, like it would get from a parent component. But I also can't assign a dummy function or Jest mock function since then onChange doesn't update the value.
I'm not sure how to fix that.
I guess if someone uses that component and doesn't give it the required props it's not the end of the world, it just won't do anything :) But for better usability they should be required.